common/workunit/workunit.cpp

```diff
@@ -8892,20 +8919,14 @@ void exportWorkUnitToXMLFileWithHiddenPasswords(IPropertyTree *p, const char *fi
 
 extern WORKUNIT_API StringBuffer &exportWorkUnitToXML(const IConstWorkUnit *wu, StringBuffer &str, bool unpack, bool includeProgress, bool hidePasswords)
 {
-    const CLocalWorkUnit *w = QUERYINTERFACE(wu, const CLocalWorkUnit);
-    if (!w)
-    {
-        const CLockedWorkUnit *wl = QUERYINTERFACE(wu, const CLockedWorkUnit);
-        if (wl)
-            w = wl->c;
-    }
-    if (w)
+    const IExtendedWUInterface *ewu = queryExtendedWU(wu);
+    if (ewu)
     {
         Linked<IPropertyTree> p;
         if (unpack||includeProgress)
-            p.setown(w->getUnpackedTree(includeProgress));
+            p.setown(ewu->getUnpackedTree(includeProgress));
         else
-            p.set(w->p);
+            p.set(ewu->queryPTree());
```

Comment:
https://track.hpccsystems.com/browse/HPCC-13621 created. I'll add a comment in the code too (but will not attempt to fix right now)